Track Buddy/MotionManager.swift

```diff
@@ -12,32 +12,57 @@ import Collections
 import CoreGraphics
 
 class MotionManager: ObservableObject {
+    private enum Parameters {
+        static let deviceMotionUpdateInterval: TimeInterval = 1/100
+        static let graphUpdateInterval = RunLoop.SchedulerTimeType.Stride(1/15)
+        static var pointStorageLimit: Int {
+            Int(1 / graphUpdateInterval.magnitude * 3)
+        } // final number after multiplier == number of seconds retained for tracer line
+    }
+    
+    // Expose graph update interval to Views that need it
+    var graphUpdateInterval: TimeInterval { Parameters.graphUpdateInterval.magnitude }
+
+    init() {
+        motionManager = CMMotionManager()
+        if motionManager.isDeviceMotionAvailable {
+            motionManager.deviceMotionUpdateInterval = Parameters.deviceMotionUpdateInterval
+            startDeviceMotion()
+        }
+        setUpThrottledPointPublisher()
+    }
+    
     private var motionManager: CMMotionManager
     
     /* With device oriented vertically, axes are:
-     +Z - braking
-     -Z - acceleration
-     +X - right
-     -X - left
+     +Z - acceleration
+     -Z - braking
+     +X - left
+     -X - right
      
      */
     
     // Acceleration values
-    @Published private(set) var x: Double = 0.0
-    @Published private(set) var y: Double = 0.0
-    @Published private(set) var z: Double = 0.0
+    private(set) var x: Double = 0.0
+    private(set) var y: Double = 0.0
+    private(set) var z: Double = 0.0
+    
     
     // Max G force values
     @Published private(set) var maxBraking: Double = 0.0
     @Published private(set) var maxAcceleration: Double = 0.0
     @Published private(set) var maxRight: Double = 0.0
     @Published private(set) var maxLeft: Double = 0.0
+
+    
+    // Rate-limited point that updates at graphUpdateInterval to smooth out graph movement
```

Comment:
Just a suggestion, you can use backticks to denote code. `graphUpdateInterval`

<img width="198" alt="Screen Shot 2021-07-01 at 12 41 08" src="https://user-images.githubusercontent.com/2940471/124180836-9d468d80-da69-11eb-923b-0c3451f051af.png">